### ISSUE TITLE
io.netty.handler.codec.EncoderException: Sending unknown packet 'serverbound/minecraft:chat_session_update'

### ISSUE THREAD

**Issue by jacobjake1** [PaperMC/Velocity#1373]
### Expected Behavior

I'm trying to connect my Velocity server which redirects me to my prison server but keep getting kicked for this error

### Actual Behavior

I get kicked for the error in the title

full error: 
[disconnect-2024-07-06_00.12.57-client.txt](https://github.com/user-attachments/files/16114935/disconnect-2024-07-06_00.12.57-client.txt)

### Steps to Reproduce

I'm trying to connect to a 1.20.4 server with 1.20.1 which was working fine but as time has gone on, I've only gotten this kick error message more often...maybe It's viaverison? but im not really sure

### Plugin List

Plugins: anarchyqueue, litebans, luckperms, viaversion, viabackwards, viarewind

### Velocity Version

Velocity 3.3.0-SNAPSHOT (git-9d25d309-b400)

**Comment by jacobjake1** [comment]
I updated to 5.0.2-SNAPSHOT+452 of ViaVerison and haven't gotten the issue since

**Comment by jacobjake1** [comment]
nevermind its still going on

**Comment by Gerrygames** [comment]
This seems to be somewhat of a vanilla client issue. The client fetches public key of the used Minecraft profile asynchronously and then sends it to the server. The key can only be updated during play state and the client does not check if it is in play state or got put into configuration state again.
However this should only really be an issue when switching servers really fast after the initial login. Might this be the case here?

**Comment by Timongcraft** [comment]
Try without Via* or contact their support, as Velocity does not support protocol hacks.

**Comment by Gerrygames** [comment]
> Try without Via* or contact their support, as Velocity does not support protocol hacks.

This probably isn't caused by ViaVersion. I suspect that the player is being sent to a different server right after connecting to the initial server. This is unsupported and the PlayerChooseInitialServerEvent should be used.

**Comment by Timongcraft** [comment]
> > Try without Via* or contact their support, as Velocity does not support protocol hacks.
> 
> This probably isn't caused by ViaVersion. I suspect that the player is being sent to a different server right after connecting to the initial server. This is unsupported and the PlayerChooseInitialServerEvent should be used.

Oh, yeah, sorry. I didn't really look at the issue but rather at the fact that they were talking a lot about ViaVersion.

**Comment by jacobjake1** [comment]
> However this should only really be an issue when switching servers really fast after the initial login. Might this be the case here?

It occurs almost always now for some reason, it could be my first connection of the day to the server and I will still get the error, I'll see if it's maybe my queue plugin and use PlayerChooseInitialServerEvent instead

**Comment by jacobjake1** [comment]
Though, a weird thing is, it only occurs on 1.21. No issues on 1.20.6 and below (server native is 1.20.4 paper)

**Comment by jacobjake1** [comment]
I ended up doing `    public void onPlayerChooseInitialServer(PlayerChooseInitialServerEvent event) {
        Player player = event.getPlayer();
        Optional<RegisteredServer> server = proxyServer.getServer(Config.serverQueue);
        server.ifPresent(event::setInitialServer);
    }` but I still get the error, so I don't think it's the queue plugin

**Comment by wordandahalf** [comment]
> This seems to be somewhat of a vanilla client issue. The client fetches public key of the used Minecraft profile asynchronously and then sends it to the server. The key can only be updated during play state and the client does not check if it is in play state or got put into configuration state again. However this should only really be an issue when switching servers really fast after the initial login. Might this be the case here?

I agree with your interpretation--we've encountered this issue, but luckily we can work around it.

**Comment by jacobjake1** [comment]
How can we work around it?

**Comment by Timongcraft** [comment]
@jacobjake1 
You just posted your whole email content. Be careful, as it contains the unsubscribe link.

**Comment by lemonlambda** [comment]
I am experiencing this but only while I'm op'ed.

**Comment by Gerrygames** [comment]
This will apparently be fixed in a future version of Minecraft: https://bugs.mojang.com/browse/MC-272506

**Comment by jacobjake1** [comment]
That's months if not years into the future, I need a way to fix it now

**Comment by electronicboy** [comment]
The only real fix we could do for velocity would be to prevent server switching until we get that packet

**Comment by jacobjake1** [comment]
If it's caused by switching servers too quickly, I could try adding a small
delay between server switches I guess

**Comment by jacobjake1** [comment]
Adding a 3 second delay before transferring the player to the main server fixed the issue. Need to make sure they are actually fully connected to the hub main server before you transition them with the delay.

**Comment by saifsuleman** [comment]
> The only real fix we could do for velocity would be to prevent server switching until we get that packet

Do you know specifically what packet this is? I'm thinking of deferring my redirects until I receive it as I'm experiencing this issue

**Comment by electronicboy** [comment]
The mojang name of the packet is in the error, don't recall off the top of my head what we call that packet

**Comment by Little100** [comment]
> Adding a 3 second delay before transferring the player to the main server fixed the issue. Need to make sure they are actually fully connected to the hub main server before you transition them with the delay.在将玩家转移到主服务器之前添加 3 秒的延迟修复了该问题。在延迟转换它们之前，需要确保它们实际上完全连接到中心主服务器。![](chrome-extension://dppcnmibpgmagiigcnfdjpnghplibbna/images/translator.png)

May I ask how you did it specifically? I'm not very familiar with Velocity. Can you tell me? I've also encountered this problem

**Comment by jacobjake1** [comment]
timeout.(3000ms) is literally all I added to my queue plugin

**Comment by ghost** [comment]
Experiencing this on 2B2T

**Comment by ShayBox** [comment]
Also experiencing it on 2B2T, CCBlueX, Hoplite, etc, but only with the ViaFabric or ViaFabricPlus mod, especially in combination with other mods that do on join server actions. Mainly a VF issue but Velocity should just make it impossible to send players to another server immediately during initial connection, forcefully delay switching a couple seconds after initially connecting to the Velocity proxy.
